gcp_variant_transforms/beam_io/vcfio.py

```diff
@@ -883,6 +884,65 @@ def expand(self, pvalue):
     return pvalue.pipeline | Read(self._source)
 
 
+def _create_vcf_source(
+    file_pattern=None, compression_type=None, allow_malformed_records=None):
+  return _VcfSource(file_pattern=file_pattern,
+                    compression_type=compression_type,
+                    allow_malformed_records=allow_malformed_records)
+
+
+class ReadAllFromVcf(PTransform):
+  """A :class:`~apache_beam.transforms.ptransform.PTransform` for reading a
+  :class:`~apache_beam.pvalue.PCollection` of VCF files.
+
+  Reads a :class:`~apache_beam.pvalue.PCollection` of VCF files or file patterns
+  and produces a PCollection :class:`Variant` (or
+  :class:`MalformedVcfRecord for failed reads) objects.
+
+  This transform should be used when reading from massive (>70,000) number of
+  files.
+  """
+
+  DEFAULT_DESIRED_BUNDLE_SIZE = 64 * 1024 * 1024  # 64MB
+
+  def __init__(
+      self,
+      min_bundle_size=0,
+      desired_bundle_size=DEFAULT_DESIRED_BUNDLE_SIZE,
+      compression_type=CompressionTypes.AUTO,
+      allow_malformed_records=False,
+      **kwargs):
+    """Initialize the :class:`ReadAllFromVcf` transform.
+
+    Args:
+      min_bundle_size (int): Minimum size of bundles that should be generated
+        when splitting this source into bundles. See
+        :class:`~apache_beam.io.filebasedsource.FileBasedSource` for more
+        details.
+      desired_bundle_size (int): Desired size of bundles that should be
+        generated when splitting this source into bundles. See
+        :class:`~apache_beam.io.filebasedsource.FileBasedSource` for more
+        details.
+      compression_type (str): Used to handle compressed input files.
+        Typical value is :attr:`CompressionTypes.AUTO
+        <apache_beam.io.filesystem.CompressionTypes.AUTO>`, in which case the
+        underlying file_path's extension will be used to detect the compression.
+      allow_malformed_records (bool): If true, malformed records from VCF files
+        will be returned as :class:`MalformedVcfRecord` instead of failing
+        the pipeline.
+    """
+    super(ReadAllFromVcf, self).__init__(**kwargs)
+    source_from_file = partial(
+        _create_vcf_source, compression_type=compression_type,
+        allow_malformed_records=allow_malformed_records)
+    self._read_all_files = filebasedsource.ReadAllFiles(
+        True, CompressionTypes.AUTO, desired_bundle_size, min_bundle_size,
```

Comment:
It is a bit unclear what `True` is without reading the source code in ReadAllFiles. I think putting it on its own line with an inline comment `# splittable` would be helpful to readers.

Reply:
Done! Good idea.